WarshipGame/Assets/Scripts/Gameplay/HexGrid.cs

```diff
@@ -0,0 +1,76 @@
+using UnityEngine;
+
+/// <summary>
+/// This script generates a Set of tiles [Width X Length] and checks if the tile can be walked on. if the tile is not passable it turns red.
+/// </summary>
+public class HexGrid : MonoBehaviour
+{
+    private GridNode[,] _grid;
+    
+    [Header("Grid Settings")] 
+    [SerializeField] private int Width = 26;
+    [SerializeField] private int Length = 26;
+    [SerializeField] private Vector2 CellSize = new Vector2(.75f, .865f);
+    [SerializeField] private Vector3 CheckBoxSize = new Vector3 (0.3f, 2f, 0.3f);
+    [SerializeField] private LayerMask ObstacleMak;
```

Comment:
ObstacleMak don't you mean ObstacleMask? if not ObstacleMak is still a bad name please change